api/v1beta1/network_disruption.go

```diff
@@ -708,11 +708,17 @@ func (s NetworkDisruptionServiceSpec) ExtractAffectedPortsInServicePorts(k8sServ
 	return goodPorts, notFoundPorts
 }
 
-// TransformCloudSpecToHostsSpec from a cloud spec disruption, get all ip ranges of services provided and transform them into a list of hosts spec
-func TransformCloudSpecToHostsSpec(cloudManager cloudservice.CloudServicesProvidersManager, cloudSpec *NetworkDisruptionCloudSpec) ([]NetworkDisruptionHostSpec, error) {
-	var hosts []NetworkDisruptionHostSpec
+// UpdateHostsOnCloudDisruption from a cloud spec disruption, get all ip ranges of services provided and appends them into the s.Hosts slice
+func (s *NetworkDisruptionSpec) UpdateHostsOnCloudDisruption(cloudManager cloudservice.CloudServicesProvidersManager) error {
+	if s.Cloud == nil {
```

Comment:
```
{"level":"error",
"ts":1704722385268.1804,
"caller":"controllers/disruption_controller.go:121",
"message":"an error occurred in reconcile loop",
"disruptionName":"46f9af52-bdf3-47d2-b19d-2e1385275bf3-r5h9k",
"disruptionNamespace":"e2e-test-3",
"error":"a panic occurred during reconcile:\n\tpanic: runtime error: invalid memory address or nil pointer dereference\n\n\terror: %!w(<nil>)",
"stacktrace":"github.com/DataDog/chaos-controller/controllers.(*DisruptionReconciler).Reconcile.func2\n\t/home/circleci/go/src/github.com/DataDog/chaos-controller/controllers/disruption_controller.go:121\nruntime.gopanic\n\t/usr/local/go/src/runtime/panic.go:920\nruntime.panicmem\n\t/usr/local/go/src/runtime/panic.go:261\nruntime.sigpanic\n\t/usr/local/go/src/runtime/signal_unix.go:861\ngithub.com/DataDog/chaos-controller/api/v1beta1.(*NetworkDisruptionSpec).UpdateHostsOnCloudDisruption\n\t/home/circleci/go/src/github.com/DataDog/chaos-controller/api/v1beta1/network_disruption.go:713\ngithub.com/DataDog/chaos-controller/controllers.(*DisruptionReconciler).startInjection\n\t/home/circleci/go/src/github.com/DataDog/chaos-controller/controllers/disruption_controller.go:492\ngithub.com/DataDog/chaos-controller/controllers.(*DisruptionReconciler).Reconcile\n\t/home/circleci/go/src/github.com/DataDog/chaos-controller/controllers/disruption_controller.go:310\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Reconcile\n\t/home/circleci/go/src/github.com/DataDog/chaos-controller/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:122\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).reconcileHandler\n\t/home/circleci/go/src/github.com/DataDog/chaos-controller/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:323\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem\n\t/home/circleci/go/src/github.com/DataDog/chaos-controller/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:274\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func2.2\n\t/home/circleci/go/src/github.com/DataDog/chaos-controller/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:235"}
```

There is a panic in the e2e tests, but I don't see how we get here with a `nil` value for `s`. Does `reflect.ValueOf(subspec).IsValid()` work differently than I expect?

Reply:
https://github.com/DataDog/chaos-controller/pull/821/files#diff-676fb126b7cd9f4e5162887cd980ec4b91cb4630cf927cae14455eca49378a9bR714

Looks like we can call this with a `nil` value for `s`.

Reply:
I'll try logging around https://github.com/DataDog/chaos-controller/pull/805/files#diff-c1d39a98823ae5bac7f016a4cf1ae0fa77de72ead1f676208a83f056125ed6a5R490-R491

Reply:
`"subspec":null,"valid":true}`
https://github.com/DataDog/chaos-controller/pull/821/files#diff-c1d39a98823ae5bac7f016a4cf1ae0fa77de72ead1f676208a83f056125ed6a5R491

okay so it literally doesn't work as i expect. I think we might want to amend that to also include a nil check

Reply:
Thanks for helping debugging those tests Philip 🙇‍♀️ it was a real struggle on my end...